src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,43 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(
+    popupSelector,
+    handleConfirmClick,
+    elementToRemove,
+    renderLoading
+  ) {
+    super(popupSelector);
+    this.submitBtn = this._popup.querySelector("#delete-btn");
+    this.element = elementToRemove;
+    this._renderLoading = renderLoading;
+    this._handleConfirmClick = () => {
+      this._renderLoading(true, this.submitBtn);
+      handleConfirmClick(this.element)
+        .then(() => {
+          this.close();
+        })
+        .catch((err) => {
+          console.error(err);
+        })
+        .finally(() => {
+          this._renderLoading(false, this.submitBtn, "Yes");
+        });
+    };
+  }
+
+  _setEventListeners() {
```

Comment:
`_setEventListeners` is not needed because you have `setEventListeners` in the parent class, which can't be called every time you open the modal: there will be `memory leak`


Just add the listener on opening liket this:

```js
  open() {
    this.submitBtn.addEventListener("click", this._handleConfirmClick);  // add it
    super.open();
  }

  close() {
    this.submitBtn.removeEventListener("click", this._handleConfirmClick);  // remove it
    super.close();
  }

```